java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -1001,6 +1005,16 @@ private static JavaType getPrimitive(JavaType primitiveOrWrapper) {
     return Optional.empty();
   }
 
+  private boolean isObjectMethod(JavaSymbol.MethodJavaSymbol methodJavaSymbol) {
+    JavaSymbol.MethodJavaSymbol overriddenSymbol = methodJavaSymbol.overriddenSymbol();
+    boolean isObjectMethod = false;
+    while (overriddenSymbol != null && !isObjectMethod) {
+      isObjectMethod = overriddenSymbol.owner.type == symbols.objectType;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)